src/screen.rs

```diff
@@ -1103,48 +1074,64 @@ impl Screen {
             need_clear_screen = true;
         }
 
-        // Output the left prompt if it has changed.
-        if self.scrolled() && !is_final_rendering {
-            self.r#move(0, 0);
-            self.write_command(ClearToEndOfLine);
-            self.actual_left_prompt = None;
-            self.actual.cursor.x = 0;
-        } else if self
-            .actual_left_prompt
-            .as_ref()
-            .is_none_or(|p| p != left_prompt)
-            || (self.scrolled() && is_final_rendering)
-            || Some(left_prompt_width) == screen_width && self.should_wrap(0)
-        {
-            self.r#move(0, 0);
-            let mut start = 0;
-            let mark_prompt_start = |zelf: &mut Screen| zelf.write_command(Osc133PromptStart);
-            if left_prompt_layout.line_breaks.is_empty() {
-                mark_prompt_start(self);
-            }
-            if self
+        let is_prompt_visible = self.desired.visible_prompt_lines > 0;
+        let prompt_last_line = self.desired.visible_prompt_lines.saturating_sub(1);
+
+        let prompt_changed = is_prompt_visible
+            && (self
                 .actual_left_prompt
                 .as_ref()
                 .is_none_or(|p| p != left_prompt)
-                || (self.scrolled() && is_final_rendering)
-            {
-                for (i, &line_break) in left_prompt_layout.line_breaks.iter().enumerate() {
+                || self.actual.visible_prompt_lines != self.desired.visible_prompt_lines);
+
+        let prompt_last_line_screen_wide =
+            is_prompt_visible && Some(prompt_last_line_width) == screen_width;
+        let prompt_last_line_should_wrap =
+            prompt_last_line_screen_wide && self.should_wrap(prompt_last_line);
+
+        // Output the left prompt if it has changed or we need to refresh softwrap.
+        if prompt_changed || prompt_last_line_should_wrap {
+            self.r#move(0, 0);
+            if self.desired.visible_prompt_lines <= 1 {
+                self.write_command(Osc133PromptStart);
```

Comment:
Re 57c79f875a1 (Try to always maintain prompt market at (0, 0), 2025-10-15)

typo in the commit message: market -> marker.